File: new_kb/parser/parser_samples/sample.lua
local Point = {}
Point.__index = Point

function Point.new(x, y)
    local self = setmetatable({}, Point)
    self.x = x
    self.y = y
    return self
end

-- adds two points
function Point:add(other)
    return Point.new(self.x + other.x, self.y + other.y)
end

local p1 = Point.new(1, 2)
local p2 = Point.new(3, 4)
print(p1:add(p2).x)

File: new_kb/parser/parser_samples/sample.scala
package com.groundwork.samples

import scala.collection.mutable.ListBuffer

case class Point(x: Int, y: Int) {
  def add(other: Point): Point = Point(x + other.x, y + other.y)
}

class Accumulator {
  private val points = ListBuffer[Point]()

  def add(p: Point): Unit = points += p

  def total: Point = points.foldLeft(Point(0, 0))(_ add _)
}

object Sample extends App {
  val acc = new Accumulator()
  acc.add(Point(1, 2))
  println(acc.total)
}
